### ISSUE TITLE
Freeze during a request

### ISSUE THREAD

**Issue by JulienPascal** [micahjsmith/FredData.jl#10]
Hi,

Thank you for this package, this is really great.
But sometimes the library gets stuck on one request. 
For instance, sometimes Julia is blocked at line 3:
```julia
using FredData
f = Fred()
FredData.get_data(f,"STTMINWGCA")
```
If I interrupt the process with Control+C and run the same command again, the pacakge works as expected. Is it because there is a limited number of requests per second associated with my API key? Is there a clean way of handling those "freezing"? So far, I have tried to add `sleep(20)` in between 2 requests, which did not solve the problem. 

Thanks.

**Comment by micahjsmith** [comment]
Thanks for reporting.

I can reproduce this by running the query in a loop and it will randomly hang after fewer than 20 iterations.

I can't reproduce this using other series IDs, and there is no documented rate limiting for the FRED API, so definitely a mystery.

Are you seeing it for just this series IDs, or for others as well?

**Comment by JulienPascal** [comment]
Hi,

I am facing this issue with more series.
I've selected a sample of the "problematic" series:
```julia
using FredData
f = Fred()

toDownload = ["STTMINWGAK", "STTMINWGAR", "STTMINWGAZ", "STTMINWGCA",
              "STTMINWGCO", "STTMINWGCT", "STTMINWGDC", "STTMINWGDE",
              "STTMINWGFG", "STTMINWGFL", "STTMINWGGA", "STTMINWGGU",
              "STTMINWGHI", "STTMINWGIA", "STTMINWGID", "STTMINWGIL",
              "STTMINWGIN", "STTMINWGKS", "STTMINWGKY"]
dataset = []
for (indexSeries, targetSeries) in enumerate(toDownload)

  println("indexSeries =  $(indexSeries)")
  println("Downloading $(targetSeries)")

  d = FredData.get_data(f, targetSeries)
  append!(dataset, [d])

  println("Done.")

end
```
The first time I run this code, I got  stuck on series n°6.
The second time, It got stuck on series n°17.
The third time, it got stuck on series n°5.

The process seems to be random.

**Comment by micahjsmith** [comment]
I looked into this but didn't respond, sorry. I was able to confirm that for certain series, such as the ones above, there will randomly be 500 second lags. There queries ultimately complete. For other series, there are no random lags.

For the same series, other clients like the Python fredapi do not hang. Thus the problem is within the julia client.

I'm not able to debug this now, anyone is welcome to chime in with findings or fixes.
